### PR TITLE
feat(deployment): single container deployment

### PR DESCRIPTION
**Pull request by miguelvr**
**AI Description**

This PR introduces the `SingleContainerDeployment` class for model deployment. It also adds the necessary imports, tests, and configurations for the new deployment option.

- **New Class**: `SingleContainerDeployment` is added to the `ModelDeploymentName` enum in `src/backend/config/deployments.py`. It extends the `BaseDeployment` class and provides methods for listing models, checking availability, and invoking chat, chat stream, and rerank functionalities.
- **Imports**: The necessary imports for `SingleContainerDeployment` are added in `src/backend/model_deployments/__init__.py`.
- **Tests**: New test files, `src/backend/tests/model_deployments/mock_deployments/mock_single_container.py` and `src/backend/tests/model_deployments/test_single_container.py`, are introduced to test the `SingleContainerDeployment` functionality.
- **Configurations**: The `SC_URL_ENV_VAR` and `SC_MODEL_ENV_VAR` environment variables are defined for the single container deployment.

### CODE REVIEW

**Comment by lusmoura** [comment]
You'll need to run `make lint` to format the files. Let me know when this is ready for review!

**Comment by miguelvr** [comment]
@lusmoura should be good to go now

**Review by lusmoura — APPROVED** [review]
LGTM!

File: src/backend/model_deployments/single_container.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+class SingleContainerDeployment(BaseDeployment):
+    """Single Container Deployment."""
+
+    client_name = "cohere-toolkit"
+
+    def __init__(self, **kwargs: Any):
+        self.url = get_model_config_var(SC_URL_ENV_VAR, **kwargs)
+        self.model = get_model_config_var(SC_MODEL_ENV_VAR, **kwargs)
+        self.client = cohere.Client(
+            base_url=self.url, client_name=self.client_name, api_key="none"
```
Comment (BeatrixCohere) [inline src/backend/model_deployments/single_container.py:32]: Should this be None?

**Review by BeatrixCohere — APPROVED** [review]

File: src/backend/model_deployments/single_container.py (hunk 1 above)
Comment (miguelvr) [reply src/backend/model_deployments/single_container.py:32]: @BeatrixCohere iirc, the client complains if there's no `api_key` set?

**Pull request merged**